pokemon-league-participation-manager/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/dao/PokemonDAOTest.java

```diff
@@ -0,0 +1,193 @@
+import cz.muni.fi.pa165.pokemon.league.participation.manager.common.PersistenceApplicationContext;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Pokemon;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.PokemonSpecies;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Trainer;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.PokemonType;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.List;
+import javax.inject.Inject;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceException;
+import javax.transaction.Transactional;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.Assert;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Unit tests for PokemonDAO implementations.
+ *
+ * @author Jiří Medveď 38451
+ */
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class PokemonDAOTest extends AbstractTestNGSpringContextTests {
+
+    @Inject
+    private PokemonDAO pokemonDao;
+
+    @Inject
+    private TrainerDAO trainerDao;
+
+    @Inject
+    private PokemonSpeciesDAO pokemonSpeciesDao;
+
+    private Trainer trainer;
+    private PokemonSpecies species;
+    private Pokemon pokemon1;
+    private Pokemon pokemon2;
+
+    @BeforeMethod
+    public void setUp() {
+        species = new PokemonSpecies();
+        species.setSpeciesName("Test species");
+        species.setPrimaryType(PokemonType.FIRE);
+        pokemonSpeciesDao.createPokemonSpecies(species);
+
+        trainer = new Trainer();
+        trainer.setName("Test trainer");
+        trainer.setBorn(LocalDate.of(2000, Month.MARCH, 1));
+        trainer.setPasswordHash("Password");
+        trainer.setSurname("Surname");
+        trainer.setUserName("Username");
+        trainerDao.createTrainer(trainer);
+
+        pokemon1 = new Pokemon();
+        pokemon1.setNickname("Pokemon 1");
+        pokemon1.setDateTimeOfCapture(LocalDateTime.of(2018, Month.MARCH, 1, 0, 0));
+        pokemon1.setSpecies(species);
+        pokemon1.setTrainer(trainer);
+        pokemonDao.createPokemon(pokemon1);
+
+        pokemon2 = new Pokemon();
+        pokemon2.setNickname("Pokemon 2");
+        pokemon2.setDateTimeOfCapture(LocalDateTime.of(2018, Month.MARCH, 1, 0, 1));
+        pokemon2.setSpecies(species);
+        pokemon2.setTrainer(trainer);
+        pokemonDao.createPokemon(pokemon2);

```

Comment:
The same thing here, maybe creating builder class looks redundant but makes more readable code, so please add.